components/org.wso2.carbon.identity.api.server.organization.user.sharing.management/org.wso2.carbon.identity.api.server.organization.user.sharing.management.v1/src/main/java/org/wso2/carbon/identity/api/server/organization/user/sharing/management/v1/core/UsersApiServiceCore.java

```diff
@@ -313,18 +405,80 @@ public UserSharedRolesResponse getSharedRoles(String userId, String orgId, Strin
+                roleDetails.setRoleName(role.getDisplayName());
+                roleDetails.setAudienceName(role.getAudience().getDisplay());
+                roleDetails.setAudienceType(role.getAudience().getType());
+                rolesList.add(roleDetails);
+            }
+        }
+        return rolesList;
+    }
+
+    /**
+     * Populates a list of UserSharedOrganizationsResponseLinks objects from the provided list of ResponseLinkDO.
+     *
+     * @param resultLinks The list of ResponseLinkDO objects to be converted.
+     * @return A list of UserSharedOrganizationsResponseLinks objects.
+     */
+    private List<UserSharedOrganizationsResponseLinks> populateUserSharedOrganizationsResponseLinks(
+            List<ResponseLinkDO> resultLinks) {
+
+        List<UserSharedOrganizationsResponseLinks> responseLinks = new ArrayList<>();
+        for (ResponseLinkDO resultLink : resultLinks) {
+            UserSharedOrganizationsResponseLinks links = new UserSharedOrganizationsResponseLinks();
+            links.href(resultLink.getHref());
+            links.rel(resultLink.getRel());
+            responseLinks.add(links);
+        }
+        return responseLinks;
+    }
+
     /**
      * Constructs a success response object for a completed process.
      *
-     * @param status  The status of the process (e.g., "Processing").
      * @param details Additional details or description about the process.
      * @return A {@link ProcessSuccessResponse} object containing the status and details of the process.
      */
-    public ProcessSuccessResponse getProcessSuccessResponse(String status, String details) {
+    private ProcessSuccessResponse getProcessSuccessResponse(String details) {
 
         ProcessSuccessResponse processSuccessResponse = new ProcessSuccessResponse();
-        processSuccessResponse.status(status);
+        processSuccessResponse.status(RESPONSE_STATUS_PROCESSING);
         processSuccessResponse.setDetails(details);
         return processSuccessResponse;
     }
+
+    /**
+     * Creates a UserSharingMgtClientException based on the provided error message.
+     *
+     * @param error The error message containing the code, message, and description.
+     * @return A UserSharingMgtClientException with the specified error details.
+     */
+    private UserSharingMgtClientException makeRequestError(UserSharingMgtConstants.ErrorMessage error) {
+
+        return new UserSharingMgtClientException(error.getCode(), error.getMessage(), error.getDescription());
+    }
+
+    /**
+     * Builds a structured error response.
+     *
+     * @param e The exception containing error details.
+     * @return An Error object containing the error code, message, description, and a trace ID.
+     */
+    private Error buildErrorResponse(UserSharingMgtException e) {
+
+        return new Error().code(e.getErrorCode()).message(e.getMessage()).description(e.getDescription())
```

Comment:
Here we are only setting the error code of the exception which will be returned to the user.

Mostly the service layer suppose to return error codes with 65xxx, or 60xxx. In such cases API layer should handle the error codes properly where it should add a condition to check whether the respective error code includes the prefix.
Without checking that condition may result in sending error codes without the Prefix.

Example: https://github.com/wso2/identity-api-server/blob/a15e96f5298fd53083b691645ef62af11e2d03b6/components/org.wso2.carbon.identity.api.server.rule.metadata/org.wso2.carbon.identity.api.server.rule.metadata.v1/src/main/java/org/wso2/carbon/identity/api/server/rule/metadata/v1/util/RuleMetadataAPIErrorBuilder.java#L33